### PR TITLE
FOUR-18065 API endpoints to encrypt/decrypt data

### PR DESCRIPTION
**Pull request by julceslauhub**
https://processmaker.atlassian.net/browse/FOUR-18065

Added next endpoints:

POST /api/1.0/encrypted_data/encryptText
POST /api/1.0/encrypted_data/decryptText

ci:next

### CODE REVIEW

File: ProcessMaker/Http/Controllers/Api/EncryptedDataController.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        // Initialize variables
+        $fieldName = $request->input('field_name');
+        $requestId = $request->input('request_id');
+        $screenId = $request->input('screen_id');
+
+        $user = Auth::user();
+        $userId = $user->id;
+
+        // Check if the user is assigned to the encrypted field
+        EncryptedData::checkUserPermission($userId, $screenId);
```
Comment (caleeli) [inline ProcessMaker/Http/Controllers/Api/EncryptedDataController.php:69]: field_name

File: ProcessMaker/Models/EncryptedData.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    protected $table = 'encrypted_data';
+
+    /**
+     * The attributes that aren't mass assignable.
+     *
+     * @var array
+     */
+    protected $guarded = [
+        'id',
+        'uuid',
```
Comment (caleeli) [inline ProcessMaker/Models/EncryptedData.php:26]: Check this section, seems uuid must be returned

**Review by caleeli — CHANGES_REQUESTED** [review]
Please review the comments, and add a feature test for the endpoints

File: ProcessMaker/Models/EncryptedData.php (hunk 2 above)
Comment (julceslauhub) [reply ProcessMaker/Models/EncryptedData.php:26]: now the endpoint only returns the uuid

**Review by caleeli — APPROVED** [review]

**Pull request merged**